index.js

```diff
@@ -0,0 +1,118 @@
+
+const utils = require('./lib/utils');
+
+module.exports = class ApiGatewayBinaryStatus {
+
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options;
+    this.provider = this.serverless.getProvider('aws');
+    this.stage = this.options.stage;
+
+    Object.assign(this,
+      utils
+    );
+
+    this.hooks = {
+      'after:aws:deploy:finalize:cleanup': this.checkBinarySupportStatus.bind(this),
+    };
+  }
+
+  checkBinarySupportStatus(){
+    this.serverless.cli.log('[serverless-plugin-apig-binary-status] Checking binary support enabled to AWS API Gateway...');
+
+    return BbPromise.resolve()
+      .bind(this)
+      .then(() => this.createTestData(this.stage))
+      .then(() => new BbPromise(resolve => setTimeout(() => resolve(), 65000)))  //replace to depend on serverless-apigwy-binary
+      .then(() => this.getFunctionsForContentHandling())
+      .then((funcs) => this.checkContentHandlingEnabled(funcs))
+      .then(() => this.deleteTestData(this.stage));
+  }
+
+  getFunctionsForContentHandling () {
+    const funcs = this.serverless.service.functions;
+    const validFuncs = {};
+
+    Object.keys(funcs).forEach((fKey) => {
+      funcs[fKey].events.forEach((e) => {
+        if (e.http && e.http.contentHandling && funcs[fKey].checkBinaryStatus) {
+          validFuncs[fKey] = funcs[fKey];
+        }
+      })
+    });
+    return validFuncs;
+  }
+
+  checkContentHandlingEnabled (funcs) {
+    if (!Object.keys(funcs).length) {
+      return;
+    }
+
+    const apigateway = new this.provider.sdk.APIGateway({
+      region: this.options.region
+    });
+
+    const lambda = new this.provider.sdk.Lambda({
+      region: this.options.region
+    });
+
+    const integrationResponse = {
+      statusCode: '200'
+    };
+
+    const apiName = this.provider.naming.getApiGatewayName();
+    const MAX_CHECK_NUM = 12;
+
+    return this.getApiGatewayResources(apigateway, apiName)
+      .then(resources => {
+        integrationResponse.restApiId = resources.restApiId;
+
+        _.mapKeys(funcs, (fValue, fKey)=>{
+          let checkCount = 0;
+
+          this.setCompressionForFunction(lambda, fValue.name, "true")
+           .then(() => {
+            let checkIfEnabled = (resources, fValue) => {
+              const results = [];
+              results.push(this.checkBinaryMediaTypes(apigateway, resources.restApiId));
+
+              _.map(fValue.events, e => {
+                if (e.http && e.http.contentHandling) {
+                  integrationResponse.httpMethod = e.http.method.toUpperCase();
+                  integrationResponse.resourceId = resources.items.find(
+                  r => r.path === `/${e.http.path}`).id;
+                  results.push(this.isContentHandlingEnabled(integrationResponse, e.http.contentHandling, apigateway));
+                }
+              })
+
+              results.push(this.testBinaryEnabled(this.stage));
+              return Promise.all(results)

```

Comment:
The `contentHandling` and `binaryMediaTypes` are set by [`serverless-apigwy-binary`](https://github.com/ryanmurakami/serverless-apigwy-binary/blob/master/index.js) and [`serverless-apigw-binary`](https://github.com/maciejtreder/serverless-apigw-binary/blob/master/index.js). To update these two settings to api gateway, have to create a new deployment: https://github.com/ryanmurakami/serverless-apigwy-binary/blob/master/index.js#L71
> Every time you update an API or its stage settings, you must redeploy the API by creating new snapshot of the API and associating the new deployment with a new or existing stage to make the changes available for your users. [refer](http://docs.aws.amazon.com/apigateway/latest/developerguide/how-to-deploy-api.html)

The race condition we saw before is that after the new deployment, there maybe a few seconds that the changes  are still not available. The way to check if the changes are available is to call the filter route to see if it responses successfully.